src/main/java/com/shapesecurity/shift/es2016/codegen/TokenStream.java

```diff
@@ -35,11 +34,8 @@ public TokenStream(@NotNull StringBuilder writer) {
     }
 
     @NotNull
-    private static String numberDot(@NotNull String fragment) {
-        if (fragment.indexOf('.') < 0 && fragment.indexOf('e') < 0) {
-            return "..";
-        }
-        return ".";
+    protected static boolean numberNeedsDoubleDot(@NotNull String fragment) {
+        return fragment.indexOf('.') < 0 && fragment.indexOf('e') < 0;
```

Comment:
This is fragile. If we start using hex/octal/binary notation or `E` in scientific notation, this will fail. Let's just check that it matches `\d+`.